internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -768,13 +770,33 @@ func NewKeystoreSecretForCR(cr *operatorv1beta1.Cryostat) *corev1.Secret {
 	}
 }
 
-func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) *corev1.ServiceAccount {
-	return &corev1.ServiceAccount{
+func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) (*corev1.ServiceAccount, error) {
+	OAuthRedirectReference := &oauthv1.OAuthRedirectReference{
+		Reference: oauthv1.RedirectReference{
+			Kind: "Route",
+			Name: cr.Name,
+		},
+	}
+
+	ref, err := json.Marshal(OAuthRedirectReference)
+	if err != nil {
+		return nil, err
+	}
+
+	sa := &corev1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      cr.Name,
 			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app": "cryostat",
+			},
+			Annotations: map[string]string{
+				"serviceaccounts.openshift.io/oauth-redirecturi.route":       "https://",
+				"serviceaccounts.openshift.io/oauth-redirectreference.route": string(ref),
```

Comment:
Question: the `serviceaccounts.openshift.io/oauth-redirecturi.route: ` from `oc describe` doesn't seem to have a URL value. How can I check that the URI has actually been generated?

Reply:
If I understand the documentation, this is the correct behaviour. In this case, the scheme provided in the `oauth-redirecturi.route` will overwrite the scheme (is there one?) determined by the route. So while the scheme comes from `oauth-redirecturi.route`, the hostname, port, and path will come from the route supplied in `oauth-redirectreference.route`.